DataFormats/PatCandidates/src/PackedCandidate.cc

```diff
@@ -11,442 +11,452 @@ CovarianceParameterization pat::PackedCandidate::covarianceParameterization_;
+    unpackCovarianceElement(*m, packedCovariance_.dptdpt, 0, 0);
+    unpackCovarianceElement(*m, packedCovariance_.detadeta, 1, 1);
+    unpackCovarianceElement(*m, packedCovariance_.dphidphi, 2, 2);
+    unpackCovarianceElement(*m, packedCovariance_.dxydxy, 3, 3);
+    unpackCovarianceElement(*m, packedCovariance_.dzdz, 4, 4);
+    unpackCovarianceElement(*m, packedCovariance_.dxydz, 3, 4);
+    unpackCovarianceElement(*m, packedCovariance_.dlambdadz, 1, 4);
+    unpackCovarianceElement(*m, packedCovariance_.dphidxy, 2, 3);
+    reco::TrackBase::CovarianceMatrix* expected = nullptr;
+    if (m_.compare_exchange_strong(expected, m.get())) {
+      m.release();
     }
-}
 
-void pat::PackedCandidate::unpackVtx() const {
-    reco::VertexRef pvRef = vertexRef();
-    dphi_ = int16_t(packedDPhi_)*3.2f/std::numeric_limits<int16_t>::max(),
-    deta_ = MiniFloatConverter::float16to32(packedDEta_);
-    dtrkpt_ = MiniFloatConverter::float16to32(packedDTrkPt_);
-    dxy_ = MiniFloatConverter::float16to32(packedDxy_)/100.;
-    dz_   = pvRef.isNonnull() ? MiniFloatConverter::float16to32(packedDz_)/100. : int16_t(packedDz_)*40.f/std::numeric_limits<int16_t>::max();
-    Point pv = pvRef.isNonnull() ? pvRef->position() : Point();
-    float phi = p4_.load()->Phi()+dphi_, s = std::sin(phi), c = std::cos(phi);
-    auto vertex = std::make_unique<Point>(pv.X() - dxy_ * s,
-                    pv.Y() + dxy_ * c,
-                    pv.Z() + dz_ ); // for our choice of using the PCA to the PV, by definition the remaining term -(dx*cos(phi) + dy*sin(phi))*(pz/pt) is zero
-
-     
-
- 
-    Point* expected = nullptr;
-    if( vertex_.compare_exchange_strong(expected,vertex.get()) ) {
-      vertex.release();
-    }
+  } else {
+    throw edm::Exception(edm::errors::UnimplementedFeature)
+        << "You do not have a valid track parameters file loaded. "
+        << "Please check that the release version is compatible with your input data"
+        << "or avoid accessing track parameter uncertainties. ";
+  }
 }
 
-pat::PackedCandidate::~PackedCandidate() { 
+void pat::PackedCandidate::unpackVtx() const {
+  reco::VertexRef pvRef = vertexRef();
+  dphi_ = int16_t(packedDPhi_) * 3.2f / std::numeric_limits<int16_t>::max(),
+  deta_ = MiniFloatConverter::float16to32(packedDEta_);
+  dtrkpt_ = MiniFloatConverter::float16to32(packedDTrkPt_);
+  dxy_ = MiniFloatConverter::float16to32(packedDxy_) / 100.;
+  dz_ = pvRef.isNonnull() ? MiniFloatConverter::float16to32(packedDz_) / 100.
+                          : int16_t(packedDz_) * 40.f / std::numeric_limits<int16_t>::max();
+  Point pv = pvRef.isNonnull() ? pvRef->position() : Point();
+  float phi = p4_.load()->Phi() + dphi_, s = std::sin(phi), c = std::cos(phi);
+  auto vertex = std::make_unique<Point>(
+      pv.X() - dxy_ * s,
+      pv.Y() + dxy_ * c,
+      pv.Z() +
+          dz_);  // for our choice of using the PCA to the PV, by definition the remaining term -(dx*cos(phi) + dy*sin(phi))*(pz/pt) is zero
```

Comment:
@davidlange6 @fabiocos 
isn't there a way in clang-format to separate the comment to be on a separate line (before or after the single-line)?
The current result looks pretty bad, breaking just a few symbols to a separate line

Reply:
i'm not aware of options for such fine grain control. Obviously this can be fixed by hand.